docs/DeveloperGuide.md

```diff
@@ -238,76 +238,85 @@ _{more aspects and alternatives to be added}_
 
 _{Explain here how the data archiving feature will be implemented}_
 
-### \[Proposed\] Mass Operations
+### Mass Operations
 
 #### What is the feature about
-This feature provides a way for users to mark and unmark multiple `Tasks` at a time. For the current implementation of
-mark and unmark, users have to type in a command for each task that they wish to mark or unmark, one at a time.
+This feature provides a way for users to `mark`, `unmark` and `delete` multiple `Tasks` at a time. For the previous implementation of `mark`, `unmark` and `delete`, users had to type in a command for each task that they wished to `mark`, `unmark` or `delete` one at a time.   
+
+`Mark` and `unmark` allows users to mark a `Task` objects as either completed or uncompleted while `delete` allows users to remove a task from the task list.
 
 #### How the feature is implemented
 
-The feature is to be implemented with the addition of a `MassOpsParser` class which parses through user inputs
-consisting of multiple indexes and processes the indexes to return an `ArrayList` of `Indexes` for `MarkCommand` and
-`UnmarkCommand` to execute on.
+These features are implemented with the addition of a `MassOpsParser` class which parses through user inputs, consisting of multiple indexes. `MassOpsParser` processes the indexes to return an `ArrayList` of `Indexes` for `MarkCommand`, `UnmarkCommand` or `DeleteCommand` to execute.
+
+It is designed to preserve the Command Design Pattern. Through the implementation of the `MarkCommmandParser` and `UnmarkCommandParser`, we can enforce the input format of mark command. Furthermore, isolating `MarkCommand` and `UnmarkCommand` into separate classes, we narrow down functionality of each class. This gives the application more control by limiting the outcome in successful execution. For example, successful execution of MarkCommand will only lead to the task being marked as complete.Whereas an alternative design combining mark and unmark functionality together will lead vague outcome (application unaware whether the task is marked as complete or incomplete after execution).
 
-##### MassOperations: Marking tasks
-Given below is an example usage scenario of how the MassOps mechanism behaves at each step to mark tasks in
+##### Mass `Marking`, `Unmarking` and `Deleting` Tasks
+Given below is an example usage scenario of how the MassOps mechanism behaves at each step to `mark` tasks in
```

Comment:
Perhaps can consider removing the heading `Mass Marking, Unmarking and Deleting Tasks`  as it seems a little redundant here